pkg/util/proxyconfig/merge.go

```diff
@@ -20,11 +20,30 @@ import (
 // provided, a comma-separated string of cluster-wide noProxy settings
 // are returned.
 func MergeUserSystemNoProxy(proxy *configv1.Proxy, infra *configv1.Infrastructure, network *configv1.Network, cluster *corev1.ConfigMap) (string, error) {
+	// TODO: This will be flexible when master machine management is more dynamic.
+	type installConfig struct {
+		ControlPlane struct {
+			Replicas string `json:"replicas"`
+		} `json:"controlPlane"`
+		Networking struct {
+			MachineCIDR string `json:"machineCIDR"`
+		} `json:"networking"`
+	}
+	var ic installConfig
+	data, ok := cluster.Data["install-config"]
```

Comment:
Do all clusters have an install-config? Including UPI clusters?

Reply:
@squeed yes, all install types include an install-config with machineCIDR. See the [user docs ](https://github.com/openshift/installer/blob/master/docs/user) as a reference. A similar installer [PR](https://github.com/openshift/installer/pull/2286) was just merged.